src/nwbinspector/_dandi_inspection.py

```diff
@@ -0,0 +1,256 @@
+        Skip the PyNWB validation step.
+        This may be desired for older NWBFiles (< schema version v2.10).
+    client: dandi.dandiapi.DandiAPIClient
+        The client object can be passed to avoid re-instantiation over an iteration.
+    """
+    if client is None:
+        import dandi.dandiapi
+
+        client = dandi.dandiapi.DandiAPIClient()
+
+    dandiset = client.get_dandiset(dandiset_id=dandiset_id, version_id=dandiset_version)
+    asset = dandiset.get_asset_by_path(path=dandi_file_path)
+    asset_url = asset.get_content_url(follow_redirects=1, strip_query=True)
+
+    yield inspect_url(
+        url=asset_url,
+        config=config,
+        checks=checks,
+        ignore=ignore,
+        select=select,
+        importance_threshold=importance_threshold,
+        skip_validate=skip_validate,
+    )
+
+
+def inspect_url(
+    *,
+    url: str,
+    config: Union[str, pathlib.Path, dict, Literal["dandi"]] = "dandi",
+    checks: Union[list, None] = None,
+    ignore: Union[List[str], None] = None,
+    select: Union[List[str], None] = None,
+    importance_threshold: Union[str, Importance] = Importance.BEST_PRACTICE_SUGGESTION,
+    skip_validate: bool = False,
+) -> Iterable[InspectorMessage]:
+    """
+    Inspect an explicit S3 URL.
+
+    Parameters
+    ----------
+    url : string
+        A URL referring to the cloud location of an NWB file.
+        Commonly used with DANDI, where the URL has a form similar to:
+        https://dandiarchive.s3.amazonaws.com/blobs/636/57e/63657e32-ad33-4625-b664-31699b5bf664
+
+        Note: this must be the `https` URL, not the 's3://' form.
+    config : file path, dictionary, or "dandi", default: "dandi"
+        If a file path, loads the dictionary configuration from the file.
+        If a dictionary, it must be valid against the configuration schema.
+        If "dandi", uses the requirements for DANDI validation.
+    checks : list, optional
+        list of checks to run
+    ignore: list, optional
+        Names of functions to skip.
+    select: list, optional
+        Names of functions to pick out of available checks.
+    importance_threshold : string or Importance, optional
+        Ignores tests with an assigned importance below this threshold.
+        Importance has three levels:
+
+            CRITICAL
+                - potentially incorrect data
+            BEST_PRACTICE_VIOLATION
+                - very suboptimal data representation
+            BEST_PRACTICE_SUGGESTION
+                - improvable data representation
+
+        The default is the lowest level, BEST_PRACTICE_SUGGESTION.
+    skip_validate : bool, default: False
+        Whether to skip the PyNWB validation step.
+    """
+    import remfile
+
+    if not isinstance(config, dict):
+        config = load_config(filepath_or_keyword=config)
+    validate_config(config=config)
+
+    byte_stream = remfile.File(url=url)
+    file = h5py.File(name=byte_stream)
+    io = pynwb.NWBHDF5IO(file=file)
```

Comment:
Would it be possible to make these contexts, or else close them properly separately?

Reply:
Mixing `yield` inside contexts can lead to strange behavior depending on how our generator is intended to be exhausted, such as in https://stackoverflow.com/questions/64352497/why-is-the-context-manager-closed-after-it-is-yielded

We can try and see what tests think

Reply:
OK that makes sense. Let's try to close the pens and if we can't let's add a comment about this

Reply:
I already implemented the contexts, tests say they run fine. Just something to keep in mind if we ever see strange behavior in the future